public/src/hooks.js

```diff
@@ -8,9 +8,38 @@ export const API = {
   axiosInstance: axios.create(axiosConfig),
   // eslint-disable-next-line no-undef
   configureUrl: (url) => `${SERVER_CONFIGURATION.baseUrl}${url}`,
-  post: (url, data) => fetch(url, data),
+  post: (url, data) => fetch(url, {
+    method: 'POST',
+    mode: 'cors',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    redirect: 'follow',
+    body: JSON.stringify(data),
+  }),
+  get: (url, data) => {
+    const params = Object.keys(data).map((key) => `${key}=${data[key]}`).join('&');
```

Comment:
You could use the URLSearchParams constructor if you're accepting modern browsers:

```js
const search = new URLSearchParams(data).toString();
```

right now you're forgetting to URI encode your key/values